import_obfuscation.go

```diff
@@ -44,28 +48,59 @@ type privateImports struct {
 	privateNames []string
 }
 
-func obfuscateImports(objPath, importCfgPath string) (map[string]string, error) {
+func appendPrivateNameMap(nameMap map[string]string, pkg string, packageDirectory string) error {
+	mapFile := filepath.Join(packageDirectory, garbleMapFile)
+	data, err := ioutil.ReadFile(mapFile)
+	if errors.Is(err, os.ErrNotExist) {
+		return nil
+	}
+
+	if err != nil {
+		return err
+	}
+
+	var localMap map[string]string
+	err = json.Unmarshal(data, &localMap)
+	if err != nil {
+		return err
+	}
+
+	for oldName, newName := range localMap {
+		nameMap[pkg+"."+oldName] = newName
+	}
```

Comment:
I don't get this; why don't you just marshal the final map into the file? you save one map, and then construct another with slightly different key strings.

Reply:
This still doesn't make sense. Why not just decode into nameMap? Remember that decoding json into a map doesn't affect previous map keys.